c_src/membrane_ffmpeg_swscale_plugin/scaler.c

```diff
@@ -196,15 +198,16 @@ UNIFEX_TERM scale(UnifexEnv *env, UnifexPayload *payload, State *state) {
   size_t payload_size = av_image_get_buffer_size(
       pixel_format, state->output_width, state->output_height, 1);
 
-  UnifexPayload *frame =
-      unifex_payload_alloc(env, UNIFEX_PAYLOAD_SHM, payload_size);
+  UnifexPayload frame;
+  unifex_payload_alloc(env, UNIFEX_PAYLOAD_SHM, payload_size, &frame);
 
   if (av_image_copy_to_buffer(
-          frame->data, payload_size, (const uint8_t *const *)state->output_data,
+          frame.data, payload_size, (const uint8_t *const *)state->output_data,
           (const int *)state->output_linesize, pixel_format,
           state->output_width, state->output_height, 1) < 0) {
     return scale_result_error(env, "copy_to_payload");
```

Comment:
We have to release frame before returning. Maybe let's write something like this
```c
  res = scale_result_error(env, "copy_to_payload);
  goto cleanup;

//...
  res = scale_result_ok(env, &frame);
cleanup:
  unifex_payload_release(&frame);
  return res;
}
```